Library/Homebrew/formula.rb

```diff
@@ -595,6 +595,14 @@ def inspect
     s << ") #{path}>"
   end
 
+  def modified?
+    old_wd = Pathname.pwd
+    Dir.chdir path.dirname
+    diff = Utils.popen_read("git", "diff", "--diff-filter=M", "--name-only", "#{path}")
+    Dir.chdir old_wd
```

Comment:
> Another thing that you need to check is formula file may be outside git repo.

That was why I added the `exitstatus` check; git should return non-zero in those cases. Are there cases where that's insufficient?